Userland/Libraries/LibPDF/Parser.h

```diff
@@ -139,7 +140,7 @@ class Parser final : public RefCounted<Parser> {
     bool consume(char);
 
     Reader m_reader;
-    RefPtr<Document> m_document;
+    WeakPtr<Document> m_document;
```

Comment:
I am not too happy about it either. I am seeing this as a temporary workaround. I agree that there is room for further improvement.
However this was the least intrusive change that I could come up with in order to fix the memory leak that became apparent when running tests with address sanitizing enabled.
If I am understanding your suggestion correctly I think this would lead to a chain of changes in order to refactor the ownership between these two classes while still having a circular dependency.
I would argue that this refactoring is outside of the scope of this PR.

Reply:
Okay! :)